api/client/service/opts.go

```diff
@@ -253,6 +307,21 @@ func (m *MountOpt) Value() []swarm.Mount {
 	return m.values
 }
 
+func getMountType(fields []string) (swarm.MountType, []string) {
```

Comment:
>  used a positional sentinel value to figure out how to parse the rest of the syntax... The complex arose when having to backtrack on that sentinel.

Right, because the sentinel wasn't explicitly a type. You had to infer the type from a value that wasn't explicit.

In this case the value is explicitly setting a type, so you would never need to backtrack. The value also can't contain the delimiter, which added to the problem.

It is not the same situation.